src/mo_frontend/typing.ml

```diff
@@ -2052,6 +2053,20 @@ and detect_lost_fields env t = function
               id
               display_typ t)
       flds
+  | ObjBlockE (_exp_opt, { it = Type.Object; _}, _typ_opt, dec_fields) ->
+    let pub_fields = pub_fields dec_fields |> snd in
+    let pub_ids = T.Env.keys pub_fields in
+    let [@warning "-8"] T.Obj (_, fts) = t in
+    List.iter
+      (fun id ->
+        match List.find_opt (fun ft -> ft.T.lab = id) fts with
+        | Some _ -> ()
+        | None ->
+           warn env ((T.Env.find id pub_fields).id_region) "M0215"
+             "Public field `%s` is ignored in object of type%a"
```

Comment:
we could just pass in the verbal context, here `"object"` or `"class"`, above `"record"` or `"record extension".